pallets/auctions/src/lib.rs

```diff
@@ -44,16 +42,114 @@ const BID_ADD_BLOCKS: u32 = 10;
 /// Minimal auction duration
 const MIN_AUCTION_DUR: u32 = 10;
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq)]
+pub enum Auction<T: Config> {
+	English(EnglishAuction<T>),
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq)]
+pub struct EnglishAuction<T: Config> {
+	pub general_data: GeneralAuctionDataOf<T>,
+	pub specific_data: EnglishAuctionData<T>,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq)]
+pub struct EnglishAuctionData<T: Config> {
+	pub reserve_price: BalanceOf<T>,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq)]
+pub struct GeneralAuctionData<AccountId, Balance, BlockNumber, NftClassId, NftTokenId, BoundedVec> {
+	pub name: BoundedVec,
+	pub last_bid: Option<(AccountId, Balance)>,
+	pub next_bid_min: Balance,
+	pub start: BlockNumber,
+	pub end: BlockNumber,
+	pub owner: AccountId,
+	pub token: (NftClassId, NftTokenId),
+}
+
 /// Define type aliases for better readability
-pub type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
-pub type AuctionInfoOf<T> = AuctionInfo<
+pub type GeneralAuctionDataOf<T> = GeneralAuctionData<
 	<T as frame_system::Config>::AccountId,
 	BalanceOf<T>,
 	<T as frame_system::Config>::BlockNumber,
 	<T as pallet_uniques::Config>::ClassId,
 	<T as pallet_uniques::Config>::InstanceId,
+	BoundedVec<u8, <T as pallet_uniques::Config>::StringLimit>,
 >;
 
+pub type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+impl<T: Config> sp_std::fmt::Debug for Auction<T> {
+	fn fmt(&self, f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		write!(f, "Auction")
+	}
+}
+pub trait NftAuction<AccountId, AuctionId, BalanceOf, NftAuction> {
+	fn bid(bidder: AccountId, auction_id: AuctionId, value: BalanceOf) -> DispatchResult;
+
+	fn close(auction: &Self) -> DispatchResult;
```

Comment:
Isn't it more clear this way ?

```suggestion
	fn close(&self) -> DispatchResult;
```